### PR TITLE
Change non-predictive toggle

### PR DESCRIPTION
**Pull request by SirSaltyy**
It's not working on my local host (the toggles get stuck and aren't reflecting the correct state). I think it's because the properties "isRanked" and "isSubsidised" isn't being written to firebase for some reason, and the default undefined state is "true".

I tried fixing it for 1 hour and I'm pretty sure it should work and it just doesn't....

Is it because it only works when live because the writes aren't actually happening in Firebase or something since I'm creating a new boolean property for the contract?

### CODE REVIEW

File: backend/shared/src/update-group-contracts-internal.ts [hunk 1]
```diff
@@ -55,12 +58,21 @@ export async function addGroupToContract(
...
-      nonPredictive: true,
+      isRanked: false,
     })
     if (recordEdit) {
-      await recordContractEdit(contract, recordEdit.userId, ['nonPredictive'])
+      await recordContractEdit(contract, recordEdit.userId, ['isRanked'])
+    }
+  }
+
+  if (group.id === UNSUBSIDIZED_GROUP_ID && !contract.isSubsidised) {
```
Comment (SirSaltyy) [inline backend/shared/src/update-group-contracts-internal.ts:70]: Or is it the fact I am just putting both these IF statements for unsubsidized and unranked in the same function and need to create 2 functions?

Comment (SirSaltyy) [reply backend/shared/src/update-group-contracts-internal.ts:70]: hmm that doesn't seem to be it either tho

File: backend/shared/src/update-group-contracts-internal.ts [hunk 2]
```diff
@@ -103,11 +115,17 @@ export async function removeGroupFromContract(
...
+    })
+    await recordContractEdit(contract, userId, ['isRanked'])
+  }
+  if (group.id === UNSUBSIDIZED_GROUP_ID && contract.isSubsidised) {
     await firestore.collection('contracts').doc(contract.id).update({
-      nonPredictive: false,
+      isSubsidised: true,
     })
-    await recordContractEdit(contract, userId, ['nonPredictive'])
+    await recordContractEdit(contract, userId, ['isRanked'])
```
Comment (IanPhilips) [inline backend/shared/src/update-group-contracts-internal.ts:128]: this should be ['isSubsidised'], correct?

File: backend/shared/src/update-group-contracts-internal.ts [hunk 3]
```diff
@@ -103,11 +115,17 @@ export async function removeGroupFromContract(
...
+  if (group.id === UNRANKED_GROUP_ID && contract.isRanked) {
+    await firestore.collection('contracts').doc(contract.id).update({
+      isRanked: true,
+    })
+    await recordContractEdit(contract, userId, ['isRanked'])
+  }
+  if (group.id === UNSUBSIDIZED_GROUP_ID && contract.isSubsidised) {
     await firestore.collection('contracts').doc(contract.id).update({
-      nonPredictive: false,
+      isSubsidised: true,
```
Comment (IanPhilips) [inline backend/shared/src/update-group-contracts-internal.ts:126]: should be false?

File: backend/shared/src/update-group-contracts-internal.ts [hunk 4]
```diff
@@ -103,11 +115,17 @@ export async function removeGroupFromContract(
       groupLinks: newLinks,
     })
 
-  if (group.id === NON_PREDICTIVE_GROUP_ID && contract.nonPredictive) {
+  if (group.id === UNRANKED_GROUP_ID && contract.isRanked) {
+    await firestore.collection('contracts').doc(contract.id).update({
+      isRanked: true,
```
Comment (IanPhilips) [inline backend/shared/src/update-group-contracts-internal.ts:120]: false here as well?

File: backend/shared/src/update-group-contracts-internal.ts [hunk 5]
```diff
@@ -55,12 +58,21 @@ export async function addGroupToContract(
       })
   }
 
-  if (group.id === NON_PREDICTIVE_GROUP_ID && !contract.nonPredictive) {
+  if (group.id === UNRANKED_GROUP_ID && !contract.isRanked) {
     await firestore.collection('contracts').doc(contract.id).update({
-      nonPredictive: true,
+      isRanked: false,
```
Comment (IanPhilips) [inline backend/shared/src/update-group-contracts-internal.ts:63]: should be true

File: backend/shared/src/update-group-contracts-internal.ts [hunk 6]
```diff
@@ -55,12 +58,21 @@ export async function addGroupToContract(
...
     })
     if (recordEdit) {
-      await recordContractEdit(contract, recordEdit.userId, ['nonPredictive'])
+      await recordContractEdit(contract, recordEdit.userId, ['isRanked'])
+    }
+  }
+
+  if (group.id === UNSUBSIDIZED_GROUP_ID && !contract.isSubsidised) {
+    await firestore.collection('contracts').doc(contract.id).update({
+      isSubsidised: false,
```
Comment (IanPhilips) [inline backend/shared/src/update-group-contracts-internal.ts:72]: should be true

File: common/src/contract.ts [hunk 7]
```diff
@@ -111,7 +111,8 @@ export type Contract<T extends AnyContractType = AnyContractType> = {
   isTwitchContract?: boolean
 
   coverImageUrl?: string
-  nonPredictive?: boolean // If true, don't include profits in leagues, etc.
+  isRanked?: boolean
+  isSubsidised?: boolean
```
Comment (IanPhilips) [inline common/src/contract.ts:115]: should be spelled `isSubsidized` - and everywhere else it's misspelled

File: backend/shared/src/update-group-contracts-internal.ts (hunk 1 above)
Comment (IanPhilips) [reply backend/shared/src/update-group-contracts-internal.ts:70]: I think you just have your cases backwards.

**Review by IanPhilips — DISMISSED** [review]
I don't understand the difference between unsubsidized and unranked but I assume you know what you're doing!

File: common/src/envs/constants.ts [hunk 8]
```diff
@@ -246,7 +246,7 @@ export const GROUP_SLUGS_TO_IGNORE_IN_MARKETS_EMAIL = [
   'manifold-features-25bad7c7792e',
   'bugs',
   'manifold-leagues',
-  'nonpredictive',
```
Comment (IanPhilips) [inline common/src/envs/constants.ts:249]: This is wrong, you're still using the nonpredictive slug here: https://manifold.markets/browse?topic=nonpredictive&t=&q=

File: common/src/envs/constants.ts [hunk 9]
```diff
@@ -322,7 +322,7 @@ export const GROUP_SLUGS_TO_HIDE_FROM_WELCOME_FLOW = [
   '-sircryptomind-crypto-stock',
   'selfresolving',
   'fun',
-  'nonpredictive',
```
Comment (IanPhilips) [inline common/src/envs/constants.ts:325]: https://manifold.markets/browse?topic=nonpredictive&t=&q=

File: common/src/envs/constants.ts (hunk 8 above)
Comment (IanPhilips) [reply common/src/envs/constants.ts:249]: If you wanted to switch the group slugs you'd have to switch it on all the denormalized data on contracts, which would not be trivial.

File: web/components/leagues/mana-earned-breakdown.tsx [hunk 10]
```diff
@@ -61,9 +61,7 @@ export const ManaEarnedBreakdown = (props: {
   const contractIds = loadingBets
     ? uniq(loadingBets.map((b) => b.contractId))
     : undefined
-  const contracts = usePublicContracts(contractIds)?.filter(
-    (c) => !c.nonPredictive
-  )
+  const contracts = usePublicContracts(contractIds)?.filter((c) => !c.isRanked)
```
Comment (IanPhilips) [inline web/components/leagues/mana-earned-breakdown.tsx:64]: this should be `c.isRanked`, not the other way around

File: common/src/envs/constants.ts (hunk 8 above)
Comment (SirSaltyy) [reply common/src/envs/constants.ts:249]: Yeah u right! Forgot I had made this change. Was planning to change the slug but then realised was easier not to and forgot to change this back. Good catch

**Pull request merged**